edb/ir/scopetree.py

```diff
@@ -439,24 +432,49 @@ def attach_path(
             #   <tuple>.bar
             #   <tuple>.bar.baz
             #
-            # This is because both link properties and tuples are
-            # *always* singletons, and so there is no semantic ambiguity
-            # as to the cardinality of the path prefix in different
-            # contexts.
+            # And Foo.bar[is Typ]@baz results in:
+            #   Foo.bar[is Typ]@baz
+            #    |-Foo.bar[is Typ]
+            #       |-Foo.bar
+            #   Foo
             #
-            # We could include other cases with invariant cardinality here,
-            # like type intersection, but we want to preserve the prefix
-            # visibility information for the sake of possible optimizations.
-            if (
-                not (is_lprop or prefix.is_linkprop_path())
-                and not prefix.is_tuple_indirection_path()
-            ):
+            # I forget why this is necessary for tuples, but it is permissable
```

Comment:
For tuples because we want 

```edgeql
select (Foo,) filter Foo.name = 'bar';
```

to work as expected (tuples are not fencing)

Reply:
That would work anyway?

Reply:
Or could easily be made to work by adjusting `_node_paths_are_not_links`. (The `name` case works anyway, but cases with links currently would break.)

Once that is fixed, two materialization tests fail without it; I'm confident that it is not necessary in principle, but I'm not sure if it is worth tracking down at the moment